File: app/src/main/java/com/akrivonos/a2chparser/retrofit/ApiRetrofitInterface.java
package com.akrivonos.a2chparser.retrofit;

import com.akrivonos.a2chparser.pojomodel.boardmodel.BoardModel;
import com.akrivonos.a2chparser.pojomodel.threadmodel.ThreadsModel;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

interface ApiRetrofitInterface {

    @GET("{nameBoard}/catalog.json")
    Call<ThreadsModel> getThreads(@Path("nameBoard") String nameBoard);

    @GET("makaba/mobile.fcgi?task=get_boards")
    Call<BoardModel> getBoardsList();
}

File: app/src/main/java/com/akrivonos/a2chparser/retrofit/ApiRetrofitInterface.kt
package com.akrivonos.a2chparser.retrofit

import com.akrivonos.a2chparser.pojomodel.boardmodel.BoardModel
import com.akrivonos.a2chparser.pojomodel.postmodel.PostModel
import com.akrivonos.a2chparser.pojomodel.threadmodel.ThreadsModel
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path

interface ApiRetrofitInterface {

    @GET("makaba/mobile.fcgi?task=get_boards")
    fun getBoards(): Single<BoardModel>

    @GET("{nameBoard}/catalog.json")
    fun getThreadsForBoard(@Path("nameBoard") nameBoard: String?): Single<ThreadsModel>

    @GET("{nameBoard}/res/{numThread}.json")
    fun getPostsForThread(@Path("nameBoard") nameBoard: String?, @Path("numThread") numThread: String?): Single<PostModel>
}
